node/forest_libp2p/src/blocksync/message.rs

```diff
@@ -22,11 +24,83 @@ pub struct BlockSyncRequest {
     pub options: u64,
 }
 
+impl BlockSyncRequest {
+    /// If a request expects blocks to be included in response.
+    pub fn include_blocks(&self) -> bool {
+        self.options == BLOCKS || self.options == BLOCKS_MESSAGES
+    }
+
+    /// If a request expects messages to be included in response.
+    pub fn include_messages(&self) -> bool {
+        self.options == MESSAGES || self.options == BLOCKS_MESSAGES
+    }
+}
+
+/// Status codes of a blocksync response.
+#[derive(Clone, Debug, PartialEq)]
+pub enum BlockSyncResponseStatus {
+    /// All is well.
+    Success,
+    /// We could not fetch all blocks requested (but at least we returned
+    /// the `Head` requested). Not considered an error.
+    PartialResponse,
+    /// Request.Start not found.
+    BlockNotFound,
+    /// Requester is making too many requests.
+    GoAway,
+    /// Internal error occured.
+    InternalError,
+    /// Request was bad
+    BadRequest,
+    /// Other undefined response code.
+    Other(i32),
+}
+
+impl Serialize for BlockSyncResponseStatus {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        use BlockSyncResponseStatus::*;
+        let code: i32 = match self {
+            Success => 1,
+            PartialResponse => 101,
+            BlockNotFound => 201,
+            GoAway => 202,
+            InternalError => 203,
+            BadRequest => 204,
+            Other(i) => *i,
+        };
+        code.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for BlockSyncResponseStatus {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let code: i32 = Deserialize::deserialize(deserializer)?;
+
+        use BlockSyncResponseStatus::*;
+        let status = match code {
+            x if x == 1 => Success,
+            x if x == 101 => PartialResponse,
+            x if x == 201 => BlockNotFound,
+            x if x == 202 => GoAway,
+            x if x == 203 => InternalError,
+            x if x == 204 => BadRequest,
```

Comment:
```suggestion
            1 => Success,
            101 => PartialResponse,
            201 => BlockNotFound,
            202 => GoAway,
            203 => InternalError,
            204 => BadRequest,
```

Reply:
oops yeah that was a lapse on my part (I suggested that).

Reply:
updated